### PR TITLE
FCBH-2882 Social provider link account on login

### PR DESCRIPTION
**Pull request by mgonzalezs**
# Description
Link social account on login
Create and link social account on login
## Issue Link
[FCBH-2882](https://fullstacklabs.atlassian.net/browse/FCBH-2882)

## How Do I QA This
Try to login with a social provider on the following 3 conditions and confirm are working
- New user
- User registered without any social account
- User registered with the social account

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed locally.

### CODE REVIEW

File: app/Http/Controllers/User/UsersController.php [hunk 1]
```diff
@@ -259,17 +259,40 @@ private function loginWithEmail($email, $password)
...
+            'provider_id'      => $provider_id,
+            'provider_user_id' => $provider_user_id,
+        ]);
 
-        return $user;
+
+        return User::with('accounts', 'profile')->whereId($user->id)->first();
     }
 
     /**
```
Comment (JramR7) [inline app/Http/Controllers/User/UsersController.php:298]: Is there a reason for this line?

Comment (mgonzalezs) [reply app/Http/Controllers/User/UsersController.php:298]: Yes, we need the response with the updated accounts and profile information

Comment (mgonzalezs) [reply app/Http/Controllers/User/UsersController.php:298]: Also removed the empty line :)

**Review by JramR7 — APPROVED** [review]
LGTM

**Pull request merged**